postgres_composite_types/__init__.py

```diff
@@ -225,6 +208,48 @@ def __new__(mcs, name, bases, attrs):
 
         return new_cls
 
+    def __init__(cls, name, bases, attrs):
+        super().__init__(name, bases, attrs)
+        if name == 'CompositeType':
+            return
+        cls.register_composite(retry=True)
+
+    def register_composite(cls, retry=False):
```

Comment:
This is the wrong place, because you might end up doing it during random steps that aren't connected to your database, i.e. it will attempt to establish a connection if it doesn't already have one. In django-postgres this would cause breakage during `docker build`.